src/store/membership_fee/actions.ts

```diff
@@ -38,13 +38,19 @@ const fieldToValidationMutation = new Map( [
 	[ 'type', SET_TYPE_VALIDITY ],
 ] );
 
-const validateFeeOnClientSide = ( fee: string ): Validity => Helper.isNonNumeric( fee ) ? Validity.INVALID : Validity.VALID;
+const validateFeeOnClientSide = ( context, fee: string ): Validity => {
+	if ( Helper.isNonNumeric( fee ) ) {
+		context.commit( SET_FEE, '' );
+		return Validity.INVALID;
+	}
+	return Validity.VALID;
+};
 
 export const actions = {
 	[ initializeMembershipFee ]( context: ActionContext<MembershipFee, any>, initialData: InitialMembershipFeeValues ) {
 		if ( initialData.fee ) {
 			context.commit( SET_FEE, initialData.fee );
-			context.commit( SET_FEE_VALIDITY, validateFeeOnClientSide( initialData.fee ) );
+			context.commit( SET_FEE_VALIDITY, validateFeeOnClientSide( context, initialData.fee ) );
```

Comment:
The `initializeMembershipFee` doesn't have unit tests at all, so I'll leave it up to you if you add some or leave that for later. It's out of scope for this ticket IMHO.